vermouth/processors/annotate_mut_mod.py

```diff
@@ -108,15 +108,52 @@ def residue_matches(resspec, residue_graph, res_idx):
     # FIXME: Once residue_graph is a digraph we can do something much much
     #        more clever, addressing arbitrarily branched polymers and
     #        termini
+    if residue_graph.degree[res_idx] == 1 and resspec.get('resname') in ['nter', 'cter']:
+        if not _terminal_matches(resspec.get('resname'), residue_graph, res_idx):
+            return False
+        # Remove resname and resid from the resspec, since they have special meaning
+        # in this case.
+        resspec = resspec.copy()
+        resspec.pop('resname', '')
+        resspec.pop('resid', 0)
+    return _subdict(resspec, residue)
+
+
+def _terminal_matches(resname, residue_graph, res_idx):
+    """
+    Tests whether the node res_idx in graph residue_graph is a cter or nter, as
+    determined by resname. Resname must be one of nter or cter.
+    It is assumed that the degree of the specified node is 1.
```

Comment:
Since this is already a seperate function you could also check and/or raise an error?!